### PR TITLE
Use array-based radix child map in radix node

### PR DESCRIPTION
**Pull request by platfowner**
Change summary:
- Replace native-map-based radix child map with array-based radix child map

Related issues: https://github.com/ainblockchain/ain-blockchain/issues/519

### CODE REVIEW

File: db/radix-child-map.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  static initChildArray() {
+    const arr = [];
+    for (let i = 0; i < 16; i++) {
+      arr.push(null);
+    }
+    return arr;
+  }
+
+  static _labelRadixToIndex(labelRadix) {
+    if (!labelRadix || !CommonUtil.isString(labelRadix)) {
```
Comment (liayoo) [inline db/radix-child-map.js:21]: redundant check? (`!labelRadix`)

File: db/radix-child-map.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    if (index >= 0 && index < 16) {
+      return this.childArray[index];
+    }
+    return null;
+  }
+
+  set(labelRadix, child) {
+    const index = RadixChildMap._labelRadixToIndex(labelRadix);
+    if (index >= 0 && index < 16) {
+      const isOverwiring = this._has(index);
```
Comment (liayoo) [inline db/radix-child-map.js:60]: typo (isOverwiring -> isOverwriting)?

File: db/radix-child-map.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+      if (this._has(index)) {
+        this.childArray[index] = null;
+        this.size--;
+        return true;
+      }
+    }
+    return false;
+  }
+
+  keys() {
```
Comment (liayoo) [inline db/radix-child-map.js:95]: For keys() and values(), do you think we can keep the insertion order? Or maybe leave as a todo?

File: db/radix-child-map.js (hunk 1 above)
Comment (platfowner) [reply db/radix-child-map.js]: Removed. Thanks!

File: db/radix-child-map.js (hunk 2 above)
Comment (platfowner) [reply db/radix-child-map.js]: Fixed. Thanks!

File: db/radix-child-map.js (hunk 3 above)
Comment (platfowner) [reply db/radix-child-map.js:95]: Good question! This order affects only the internal order of radix children, and the external list is generated by the StateNode.getChildLabels() and StateNode.getChildNodes() which are controlled by RadixTree.labels() and RadixTree.stateNodes(), respectively. So far, an extra hash map (RadixTree.stateNodeMap) is used internally so the order is kept to be the insertion order. I have a plan to disable the use of the extra hash map. In the case, as you mentioned we need to think of how to handle the child order issue. I'll take a look. Thanks.

**Comment by platfowner** [comment]
Comments addressed. PTAL~ @liayoo

File: db/radix-child-map.js (hunk 3 above)
Comment (liayoo) [reply db/radix-child-map.js:95]: Gotcha! Thanks for the explanation!

**Review by liayoo — APPROVED** [review]
LGTM, thanks!

**Pull request merged**